common/src/main/java/com/redhat/cloud/notifications/models/dto/v1/endpoint/EndpointDTO.java

```diff
@@ -0,0 +1,170 @@
+package com.redhat.cloud.notifications.models.dto.v1.endpoint;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.CamelPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.EndpointPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.SystemSubscriptionPropertiesDTO;
+import com.redhat.cloud.notifications.models.dto.v1.endpoint.properties.WebhookPropertiesDTO;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public final class EndpointDTO {
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    private UUID id;
+
+    @JsonProperty(required = true)
+    @NotNull
+    @Size(max = 255)
+    private String name;
+
+    @JsonProperty(required = true)
+    @NotNull
+    private String description;
+
+    private Boolean enabled = Boolean.FALSE;
+
+    private EndpointStatusDTO status = EndpointStatusDTO.UNKNOWN;
+
+    @JsonProperty(required = true)
+    @Min(0)
+    private int serverErrors;
+
+    @JsonProperty(required = true)
+    @NotNull
+    private EndpointTypeDTO type;
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonProperty(required = true)
+    @Size(max = 20)
+    private String subType;
+
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    private LocalDateTime created;
```

Comment:
`created` has an additional Jackson annotation in `CreationTimestamped`. Do we need it here?

Reply:
Well, I have tried sending a request with and without the `@JsonFormat` annotation and I haven't seen much of a difference, but let's move it to the DTO and keep it just in case. We can always remove some of these in a V2.